plugins/destination/plugin_testing_write_append.go

```diff
@@ -40,7 +40,6 @@ func (s *PluginTestSuite) destinationPluginTestWriteAppend(ctx context.Context,
 		MaxRows:    1,
 	}
 	record1 := testdata.GenTestData(mem, table.ToArrowSchema(), opts)[0]
-	record1.Retain()
```

Comment:
Why this is removed? I specifically called `Retain` here because unlike in the server we do use this after call to `writeOne` or `writeAll` for compare/testing purposes. This is also why I called `defer record1.Release` afterwards.

Reply:
This is basically just a refactor: no change in functionality. Rather than have many retains, I think it's better to have it only in `writeAll` (and by extension `writeOne`), where it's clear that the resource is being sent over a channel. Here, just by looking at the call to `writeOne`, you can't really know whether it's going to be sent over a channel or not.

The record is still available for comparison until the end of this function because the release is deferred.

Reply:
I think it is crucial for it to be here. I can explain :) maybe let's sync tmrw morning.